Common/cpp/SharedItems/Shareables.cpp

```diff
@@ -67,30 +67,43 @@ jsi::Value RetainingShareable::getJSValue(jsi::Runtime &rt) {
     return toJSValue(rt);
   } else if (remoteValue == nullptr) {
     value = toJSValue(rt);
-    remoteValue = std::make_shared<jsi::WeakObject>(rt, value.asObject(rt));
+    remoteValue = new jsi::WeakObject(rt, value.asObject(rt));
   } else {
     value = remoteValue->lock(rt);
     if (value.isUndefined()) {
       value = toJSValue(rt);
-      remoteValue = std::make_shared<jsi::WeakObject>(rt, value.asObject(rt));
+      delete remoteValue;
+      remoteValue = new jsi::WeakObject(rt, value.asObject(rt));
     }
   }
   return value;
 }
 
+RetainingShareable::~RetainingShareable() {
+  if (!runtimeHelper_->uiRuntimeDestroyed) {
+    delete remoteValue;
+  }
```

Comment:
This is expected to leak on reloads, right? If yes, can we document it here in a comment?